lib/core/database/database.dart

```diff
@@ -38,3 +40,46 @@ LazyDatabase _openConnection() {
     return NativeDatabase.createInBackground(file);
   });
 }
+
+Future<String?> exportDatabase() async {
+  final Directory dbFolder = await getApplicationDocumentsDirectory();
+  final File file = File(join(dbFolder.path, 'thunder.sqlite'));
+
+  return await FlutterFileDialog.saveFile(
+    params: SaveFileDialogParams(
+      mimeTypesFilter: ['application/octet-stream'],
+      sourceFilePath: file.path,
+      fileName: 'thunder.sqlite',
+    ),
+  );
+}
+
+Future<bool> importDatabase() async {
+  final String? filePath = await FlutterFileDialog.pickFile(
+    params: const OpenFileDialogParams(
+      fileExtensionsFilter: ['sqlite'],
+    ),
+  );
+
+  if (filePath != null) {
+    final Directory dbFolder = await getApplicationDocumentsDirectory();
+    final File file = File(join(dbFolder.path, 'thunder.sqlite'));
+
+    try {
+      // Read the selected db file
+      final List<int> bytes = await File(filePath).readAsBytes();
+
+      // Write the file out the location we read it from
+      await file.writeAsBytes(bytes, flush: true);
```

Comment:
While this works, we might want to think about doing this the way drift documentation suggests!

https://drift.simonbinder.eu/docs/examples/existing_databases/#exporting-a-database

Reply:
Alrighty, I gave this a shot, but it didn't quite work the way I was hoping. The Drift documentation assumes you already have a path, but of course we need to prompt the user for a path using the save dialog. I found some weird behavior from `flutter_file_dialog` where the save file path didn't correspond to a real path on the device, so Drift could find the value to save into.

I even tried the `file_picker` package (which seems to be more popular), and it had a slightly different but similar issue where the path it returned from saving a file was a real path, but not the one I picked.

Maybe the path I'm picking is hitting an edge case in these packages, but at least with the originally implementation, `flutter_file_dialog` handles both picking the file path, and also transferring the data internally, so it doesn't matter that it returns an unusual path.

What do you think?

Reply:
Ahh okay, that's alright - we'll roll with this for now then! Thanks for trying it out though 😄